cub/cub/device/dispatch/tuning/tuning_radix_sort.cuh

```diff
@@ -18,240 +18,682 @@
-template <typename ValueT, size_t KeySize, size_t ValueSize, size_t OffsetSize>
-struct sm100_small_key_tuning : sm90_small_key_tuning<KeySize, ValueSize, OffsetSize>
-{};
+#if !_CCCL_COMPILER(NVRTC)
+  friend ::std::ostream& operator<<(::std::ostream& os, const delay_constructor_policy& p)
+  {
+    return os << "delay_constructor_policy { .kind = " << p.kind << ", .delay = " << p.delay
+              << ", .l2_write_latency = " << p.l2_write_latency << " }";
+  }
+#endif // !_CCCL_COMPILER(NVRTC)
+};
 
-// clang-format off
+// TODO(bgruber): this is modeled after <look_back_helper.cuh>, unify this
+template <delay_constructor_kind Kind, unsigned int Delay, unsigned int L2WriteLatency>
+struct __delay_constructor_t_helper
+{
+private:
+  using delay_constructors = ::cuda::std::__type_list<
+    detail::no_delay_constructor_t<L2WriteLatency>,
+    detail::fixed_delay_constructor_t<Delay, L2WriteLatency>,
+    detail::exponential_backoff_constructor_t<Delay, L2WriteLatency>,
+    detail::exponential_backoff_jitter_constructor_t<Delay, L2WriteLatency>,
+    detail::exponential_backoff_jitter_window_constructor_t<Delay, L2WriteLatency>,
+    detail::exponential_backon_jitter_window_constructor_t<Delay, L2WriteLatency>,
+    detail::exponential_backon_jitter_constructor_t<Delay, L2WriteLatency>,
+    detail::exponential_backon_constructor_t<Delay, L2WriteLatency>>;
+
+public:
+  using type = ::cuda::std::__type_at_c<static_cast<int>(Kind), delay_constructors>;
+};
 
-// keys
+template <delay_constructor_kind Kind, unsigned int Delay, unsigned int L2WriteLatency>
+using delay_constructor_t = typename __delay_constructor_t_helper<Kind, Delay, L2WriteLatency>::type;
 
-// same as previous tuning
-template <typename ValueT> struct sm100_small_key_tuning<ValueT, 1,  0, 4> : sm90_small_key_tuning<1, 0, 4> {};
+struct radix_sort_histogram_policy
+{
+  int block_threads;
+  int items_per_thread;
+  int num_parts;
+  int radix_bits;
 
-// ipt_20.tpb_512 1.013282  0.967525  1.015764  1.047982
-// todo(@gonidelis): insignificant performance gain, need more runs.
-template <typename ValueT> struct sm100_small_key_tuning<ValueT, 2,  0, 4> { static constexpr int threads = 512; static constexpr int items = 20; };
+  _CCCL_API constexpr friend bool
+  operator==(const radix_sort_histogram_policy& lhs, const radix_sort_histogram_policy& rhs)
+  {
+    return lhs.block_threads == rhs.block_threads && lhs.items_per_thread == rhs.items_per_thread
+        && lhs.num_parts == rhs.num_parts && lhs.radix_bits == rhs.radix_bits;
+  }
 
-// ipt_21.tpb_512 1.002873  0.994608  1.004196  1.019301
-// todo(@gonidelis): insignificant performance gain, need more runs.
-template <typename ValueT> struct sm100_small_key_tuning<ValueT, 4,  0, 4> { static constexpr int threads = 512; static constexpr int items = 21; };
+  _CCCL_API constexpr friend bool
+  operator!=(const radix_sort_histogram_policy& lhs, const radix_sort_histogram_policy& rhs)
+  {
+    return !(lhs == rhs);
+  }
```

Comment:
Nitpick: We could guard all `operator!=` on `#if _CCCL_STD_VER <= 2017`

Reply:
Or we just wait until C++20 and refactor.